### PR TITLE
fix: [SC-10740] Forward disabledDays prop to DateField mock for testing

### PR DESCRIPTION
**Pull request by blambillotte**

### CODE REVIEW

**Review by ksk5280 — APPROVED** [review]
👍

**Review by bdow — APPROVED** [review]
When I first saw this I was thinking its too bad to add JSON into the markup just for testing purposes, but then realized this is just the mock example of the date field, so hooray! Glad we don't have to shove stuff into the actual application and instead can just change the mock.

**Review by stephenh — APPROVED** [review]

File: src/inputs/DateField.mock.tsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ export function DateField(props: DateFieldProps) {
       onFocus={() => maybeCall(onFocus)}
       disabled={!!props.disabled}
       readOnly={props.readOnly}
+      data-disabled-days={JSON.stringify(props.disabledDays)}
```
Comment (KoltonG) [inline src/inputs/DateField.mock.tsx:25]: **praise:** Neat!

File: src/inputs/DateField.mock.test.tsx [hunk 2]
```diff
@@ -33,4 +33,12 @@ describe("MockDateField", () => {
     expect(onFocus).toBeCalledTimes(1);
     expect(onBlur).toBeCalledTimes(1);
   });
+
+  it("forwards disabledDays prop as stringified object", async () => {
+    const testDate = new Date(2020, 0, 1)
+    const r = await render(<MockDateField label="Start Date" value={new Date(2020, 0, 1)} disabledDays={{ before: testDate }}  onChange={() => {}} />);
+
+    const expectedResult = JSON.stringify({ before: testDate });
+    expect(r.date()).toHaveAttribute("data-disabled-days", expectedResult);
```
Comment (KoltonG) [inline src/inputs/DateField.mock.test.tsx:42]: **praise:** Pretty neat!

**Review by KoltonG — APPROVED** [review]
🍣 thank you!!

**Review by chr1sjf0x — APPROVED** [review]
🍣

File: src/inputs/DateField.mock.test.tsx [hunk 3]
```diff
@@ -33,4 +33,12 @@ describe("MockDateField", () => {
     expect(onFocus).toBeCalledTimes(1);
     expect(onBlur).toBeCalledTimes(1);
   });
+
+  it("forwards disabledDays prop as stringified object", async () => {
+    const testDate = new Date(2020, 0, 1)
+    const r = await render(<MockDateField label="Start Date" value={new Date(2020, 0, 1)} disabledDays={{ before: testDate }}  onChange={() => {}} />);
```
[the remark this answers is not in the document]
Comment (blambillotte) [reply src/inputs/DateField.mock.test.tsx]: Good catch! That was dumb of me 🤣

**Pull request merged**